### DISCUSSION TITLE
Directus on multi core server

### DISCUSSION THREAD

**Discussion by samehhady** [directus/directus#18920] [Feature Requests]
### Summary

So, I am trying to find the best usage of Directus on multi core environment, by default NodeJs will use single core and will not utilize other available cores.
I see 2 approaches to take

- Use AWS load balancers and host Directus on a task with 1 vCPU only and scale it horizontally.
- Host Directus on multi core instance and use **pm2-runtime** and run Directus as cluster for all available cores.
- Use both, run with load balancers with multi core and use **pm2-runtime**

All approaches would work fine but my only issue is in all cases Directus will only use 1 core for a single processing, I have a heavy filter call that returns lots of data and takes around 30 seconds to complete, this call will only use 1 vCPU whatever the approach I am going to take.

Is there a way to split the work on this heavy processing call to multi cpus? is there a better approach than the ones above?
Can this be implemented in Directus itself as a cluster mode? and when we call the endpoints we actually split the work?

### Motivation

I want to utilize all server resources while calling Directus, I don't want to endup with a server with 4 cores and we only use the power of 1

### Detailed Design

I have Directus running on AWS ECS, I have load balancers and able to scale horizontally or vertically

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

I don't see really any drawback of handling clusters from Directus perspective.

### Alternatives

Load balancers

### Adoption Strategy

I guess only extra env variable to mention it as cluster.

**Comment by rijkvanzanten** [comment]
NodeJS / JavaScript in general are designed around a single-thread event-loop. Making that setup multi-threaded is no small feat 👀 I'd normally recommend horizontally scaling across multiple 1vCPU containers 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
